### PR TITLE
Update doc to include Benchmark Dashboard

### PR DESCRIPTION
**Pull request by guangy10**
- Linked to the dashboard
 - Removed static table
 - Updated supported backends to include MPS

### CODE REVIEW

**Review by kirklandsign — APPROVED** [review]

File: extension/benchmark/README.md [hunk 1]
```diff
@@ -17,35 +17,25 @@ The ExecuTorch project introduces an advanced benchmarking infrastructure design
...
+**Metric Tracking**:
+- Monitor essential metrics such as load time and inference time for different model implementations.
+- For LLMs, additional metrics like tokens/s are available.
+- Observe performance trends over time to identify improvements or regressions.
+
+**Visualizations**:
+- View detailed performance data through charts and graphs.
+- Color-coded highlights for improvements (green) and regressions (red) exceeding 5% compared to the baseline.
+
+The dashboard is refreshed nightly with new models, metrics, and features to provide the most current and relevant information for ExecuTorch performance benchmarking across various model types and configurations.
```
[the remark this answers is not in the document]
Comment (kirklandsign) [reply extension/benchmark/README.md:38]: Interesting dashboard. The variance is a little bit larger than what i expected.

[the remark this answers is not in the document]
Comment (guangy10) [reply extension/benchmark/README.md:38]: variance?

[the remark this answers is not in the document]
Comment (kirklandsign) [reply extension/benchmark/README.md:38]: For example, 
this row
```
 dl3 xnnpack (q8)
Samsung Galaxy S22 Ultra 5G (Android 12)
1093.5424736999998 → 723.2906561
0
251.22651 → 200.892239
```
This is variance from noice. We don't expect significant change between 10/11 and 10/13

[the remark this answers is not in the document]
Comment (guangy10) [reply extension/benchmark/README.md:38]: Yeah, btw how many iterations are we running? Maybe we should consider increasing the iterations later?

[the remark this answers is not in the document]
Comment (kirklandsign) [reply extension/benchmark/README.md:38]: Right now 10 iter

[the remark this answers is not in the document]
Comment (huydhn) [reply extension/benchmark/README.md:38]: The dashboard has data from both nightly run from main and PR.  The former is useful to see trend and catch day-to-day regression, while the latter is usually to show experiment results.

File: .lintrunner.toml [hunk 2]
```diff
@@ -151,7 +151,8 @@ command = [
     'lintrunner_adapters',
     'run',
     'grep_linter',
-    '--pattern= Executorch\W+',
```
Comment (dbort) [inline .lintrunner.toml:154]: I think the space here may be important. Though `\bExecutorch` may be more of what was intended.

**Review by huydhn — APPROVED** [review]

**Comment by guangy10** [comment]
@kirklandsign maybe we should increase the iterations to offset the data fluctuation we observed on the dashboard. Checked with @shoumikhin , the iOS side is running 5 iterations atm. I think we should bump up both, maybe to 50?

**Pull request closed without merging**